docs/Guide.Architecture.md

```diff
@@ -0,0 +1,17 @@
+# Architecture Deep Dive
```

Comment:
This is less of an architecture overview and more of a "How Detox Works" document, it is also not really a guide.
After going through the docs again, I think it best fits as a subsection/example in Introduction.HowDetoxWorks.md